### ISSUE TITLE
Add GUI to Home Assistant

### ISSUE THREAD

**Issue by pepsonEL** [leshqo/satel_integra_ext-ha#5]
Hi. Good feature will be add GUI to Home Assistant to setup integration.

**Comment by michaljonik** [comment]
I agree it's nice to have but you normally configure it just once and forget about it.
Unless you have some other use-cases where the GUI could be really a game changer?

**Comment by pepsonEL** [comment]
Yes you are right, but more integration setup once and forgot, and many integration get GUI in Home Assistant.

**Comment by michaljonik** [comment]
Home Assistant config flows seem to be quite limited. It only supports forms with flat list of entries. No multiple fields in a row. No fancy features. And satel has lots of inputs/outputs which makes it a bit tricky.

One solution would be to split in multiple very long forms like these below. Non-empty names would enable items. Items with empty names would be disabled. Other attributes (like icons) would have to be modified via normal entity customizations.

- Main settings
  - host
  - port
- Partitions names
  - partition 1
  - partition 2
  - ....
  - partition 31
  - partition 32
- Inputs names
  - input 1
  - input 2
  - ....
  - input 255
  - input 256
- Outputs names
  - output 1
  - output 2
  - ...
  - output 255
  - output 256
- Switchable outputs names
  - output 1
  - output 2
  - ...
  - output 255
  - output 256
- Temperature sensors names
  - sensor 1 name
  - sensor 2 name
  - ...
  - sensor 255
  - sensor 256

Any feedbacks or suggestions to improve this idea are welcome.

**Comment by mateusz-peczkowski** [comment]
GUI is nice to have. However, regarding the definition of partitions, switches etc., I tested openHAB and satel integration there. There is auto-discovery that detects Satel configuration... As I remember it was like I had to configure "bridge" details and then click "discover" and it crawled for me all settings from satel... Maybe we can achieve something similar here?

**Comment by michaljonik** [comment]
I had this idea in mind.

At least we could read the Satel model (available via the Satel API) which implies the number of inputs/outputs/zones and basically expose them all and let a user customize the entities (name, icon, class, disable, etc).

A more sophisticated approach would be to read out all the names from Satel (one by one), set the entities names and disable unused ones by default.

**Comment by mateusz-peczkowski** [comment]
yeah. But it will be hard to manage on a popup base like other integrations have. If you have Satel with 256 inputs/outputs, it will be 512 rows at least + partitions... So rather, it should scan, add all of them (try to get input/output name from Satel if available), and then you can edit the name/id in the entities table if you wish. It is still better than adding manually to yamls for beginners. Question is just what will happen if you will for example add new output in satel. It should pull it OR manually add

**Comment by michaljonik** [comment]
a few random thoughts ;)

* for some users it's easier in the GUI, but for others it's easier in YAML (especially if you have 256 inputs/outputs)
* AFAIK Satel API does not provide an information about input and output types, so changing between an output (binary_sensor in HA) and a switchable output (switch in HA) has to be done manually somehow
* also I think the API does not give information whether an input/output/partion is beeing used or not, which means there is no such think as "add new output it satel" ;)

**Comment by mateusz-peczkowski** [comment]
Hmmm.. You are right, BUT in my case (at least), not used inputs/outputs don't have a defined name. So maybe we can import if the name is not an empty string?

Is there an option that auto-discovery will print YAML file that you can modify whenever you want? Maybe this will be a case

**Comment by Tommatheussen** [comment]
I'd like to mention that the core integration now features UI setup. I also thought about the auto discovery, by just grabbing all inputs/outputs etc and see if they have a name defined. I haven't tried but I think it might be a big overhead on performance during setup, it's something I could try testing in the future...
